File: clients/client-python/tests/test_client.py
import unittest

from vexilla_client import VexillaClient

class TestClient(unittest.TestCase):

  __uuid = "b7e91cc5-ec76-4ec3-9c1c-075032a13a1a"

  def test_client(self):
    """
    Test the client for working and non working gradual values
    """

    client = VexillaClient(
      "https://streamparrot-feature-flags.s3.amazonaws.com",
      "dev",
      self.__uuid
    ).fetch_flags("features")

    self.assertTrue(client.should("testingWorkingGradual"))
    self.assertFalse(client.should("testingNonWorkingGradual"))

if __name__ == '__main__':
    unittest.main()

File: clients/js/packages/client-axios/src/test_client.ts
import { VexillaClient } from "./client";

(async () => {
  let uuid = "b7e91cc5-ec76-4ec3-9c1c-075032a13a1a";

  const client = new VexillaClient({
    baseUrl:
      "http://streamparrot-feature-flags.s3-website-us-east-1.amazonaws.com",
    environment: "dev",
    customInstanceHash: uuid,
  });

  const flags = await client.getFlags("features.json");

  client.setFlags(flags);

  const shouldGradual = client.should("testingWorkingGradual");
  const shouldNotGradual = client.should("testingNonWorkingGradual");

  if (!shouldGradual) {
    console.error("Should Gradual, but couldn't");
    process.exit(1);
  }

  if (shouldNotGradual) {
    console.error("Should Not Gradual, but could");
    process.exit(1);
  }
})();
